test/src/Provider/AbstractProviderTest.php

```diff
@@ -185,6 +193,33 @@ public function getHeadersTest()
         $this->assertEquals(['Authorization' => 'Bearer xyz'], $provider->getHeaders($token));
     }
 
+    public function testRandomGeneratorCreatesRandomState()
+    {
+        $xstate = str_repeat('x', 32);
+
+        $generator = m::mock('RandomLib\Generator');
+        $generator->shouldReceive('generateString')->with(32)->times(1)->andReturn($xstate);
+
+        $factory = m::mock('RandomLib\Factory');
+        $factory->shouldReceive('getMediumStrengthGenerator')->times(1)->andReturn($generator);
+
+        $provider = new MockProvider([], ['randomFactory' => $factory]);
+
+        $url = $provider->getAuthorizationUrl();
+
+        parse_str(parse_url($url, PHP_URL_QUERY), $qs);
+
+        $this->assertArrayHasKey('state', $qs);
+        $this->assertSame($xstate, $qs['state']);
+
+        // Same test, but using the non-mock implementation
+        $url = $this->provider->getAuthorizationUrl();
+
+        parse_str(parse_url($url, PHP_URL_QUERY), $qs);
+
+        $this->assertRegExp('/[a-zA-Z0-9\/+]{32}\b/', $qs['state']);
```

Comment:
@SammyK part of the issue is that it can't be easily tested... the generated string is random, so there's no guarantee we'll get a string that contains `+` or `/`. And it's also not a proper unit, because it is testing a separate method `httpBuildQuery`.